pkg/build/builder/docker.go

```diff
@@ -147,16 +146,18 @@ func (d *DockerBuilder) Build() error {
 
 	if push && pushTag != "" {
 		// Get the Docker push authentication
-		pushAuthConfig, authPresent := dockercfg.NewHelper().GetDockerAuth(
-			pushTag,
-			dockercfg.PushAuthType,
-		)
-		if authPresent {
-			log.V(4).Infof("Authenticating Docker push with user %q", pushAuthConfig.Username)
+		searchPaths := dockercfg.NewHelper().GetDockerAuthSearchPaths(dockercfg.PushAuthType)
+		ref, err := reference.Parse(pushTag)
+		if err != nil {
+			log.V(0).Infof("error parsing image name %s: %v", ref, err)
+		}
+		if ref.Registry == "" {
```

Comment:
(The `reference.Parse`+`"docker.io"` default calls could be moved insider `reportPushFailure`, the only consumer.)

Reply:
getting some very odd unit test errors on this one ... gonna punt on it